### PR TITLE
[consensus][reconfig] extend reconfiguration test for a few epochs

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

We're able to test reconfiguration flow in unit test stacked with the recovery flow changes.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Added test.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: consensus/src/chained_bft/chained_bft_smr.rs [hunk 1]
```diff
@@ -127,6 +127,7 @@ impl<T: Payload> ChainedBftSMR<T> {
                         event_processor = epoch_manager.start_new_epoch(ledger_info);
                         // clean up all the previous messages from the old epochs
                         network_receivers.clear_prev_epoch_msgs();
+                        event_processor.start().await;
```
Comment (dmitri-perelman) [inline consensus/src/chained_bft/chained_bft_smr.rs]: Lol duh

**Comment by davidiw** [comment]
Curious how this could work considering that SafetyRules would be confused by the resetting of the the round to 0 after an epoch change :O

**Comment by davidiw** [comment]
Alright, I made safety rules play nice with epoch changes as well as that the epochs really begin at 1. This caused a few other tests to require tweaks.
See: https://github.com/libra/libra/pull/1615

**Pull request closed without merging**